radixdlt/src/main/java/com/radixdlt/consensus/safety/SafetyState.java

```diff
@@ -18,41 +18,55 @@
 package com.radixdlt.consensus.safety;
 
 import com.google.inject.Inject;
-import com.radixdlt.consensus.Round;
+import com.radixdlt.consensus.QuorumCertificate;
+import com.radixdlt.consensus.View;
 
 import java.util.Objects;
+import java.util.Optional;
 
 /**
  * The state maintained to ensure the safety of the consensus system.
  */
 final class SafetyState {
-	final Round lastVotedRound; // the last round this node voted on
-	final Round lockedRound; // the highest 2-chain head
+	private final View lastVotedView; // the last view this node voted on (and is thus safe)
+	private final View lockedView; // the highest 2-chain head
+	private final View committedView; // the highest 3-chain (executed) head
+	private final QuorumCertificate genericQC; // the highest 1-chain head
 
 	@Inject
 	protected SafetyState() {
-		this(Round.of(0L), Round.of(0L));
+		this(View.of(0L), View.of(0L), View.of(0L), null);

```

Comment:
I think these should generally be `View.genesis()` instead of `View.of(0L)`

Reply:
Yeah, good idea